src/java/org/apache/cassandra/db/ClusteringComparator.java

```diff
@@ -222,6 +224,89 @@ public void validate(ClusteringPrefix clustering)
         }
     }
 
+    /**
+     * Produce a prefix-free byte-comparable representation of the given value, i.e. such a sequence of bytes that any
+     * pair x, y of valid values of this type
+     *   compare(x, y) == compareLexicographicallyUnsigned(asByteComparable(x), asByteComparable(y))
+     * and
+     *   asByteComparable(x) is not a prefix of asByteComparable(y)
+     */
+    public ByteComparable asByteComparable(ClusteringPrefix clustering)
+    {
+        return new ByteComparableClustering(clustering);
+    }
+
+    /**
+     * A prefix-free byte-comparable representation for a clustering or prefix.
```

Comment:
What exactly is meant by `prefix-free` here? I understand what the words mean but it is unclear to me how this impacts the result.

Reply:
Yes it does, thankyou